scripts/ci/travis/run-tests.sh

```diff
@@ -19,56 +22,262 @@ esac
+mkdir -p "$WORK_DIR"
+
+
+# Test step definitions
+
+
+_install_chainer_deps() {
+    # Install extras_require of Chainer.
+    local extras="$1"
+
+    # It's not possible to install only requirements.
+    # Chainer is uninstalled after the installation.
+    # TODO(niboshi): Use other installation tool
+    pip install -e "$REPO_DIR"["$extras"]
+    pip uninstall -y chainer
+}
+
+
+step_install_chainer_style_check_deps() {
+    _install_chainer_deps stylecheck
+}
+
+
+step_install_chainer_test_deps() {
+    _install_chainer_deps test
+
+    # Optional packages
+    local reqs=(
+        theano
+        h5py
+        pillow
+    )
+    pip install "${reqs[@]}"
+}
+
+
+step_before_install_chainermn_test_deps() {
+    case "$TRAVIS_OS_NAME" in
+        linux)
+            local pkgs=(openmpi-bin openmpi-common libopenmpi-dev)
+            sudo apt-get install -y "${pkgs[@]}"
+            ;;
+        osx)
+            brew install open-mpi
+            ;;
+        *)
+            false
+            ;;
+    esac
+}
+
+
+step_install_chainermn_test_deps() {
+    pip install mpi4py
+}
+
+
+step_install_chainer_docs_deps() {
+    _install_chainer_deps docs
+}
+
+
+step_python_style_check() {
+    check_targets=(
+        "$REPO_DIR"/*.py
+        "$REPO_DIR"/chainer
+        "$REPO_DIR"/chainermn
+        "$REPO_DIR"/chainerx
+        "$REPO_DIR"/tests
+        "$REPO_DIR"/examples
+        "$REPO_DIR"/chainerx_cc/examples
+    )
+
+    # Check all targets exist
+    for f in "${check_targets[@]}"; do test -e "$f"; done
+
+    flake8 --version
+    flake8 "${check_targets[@]}"
+
+    autopep8 --version
+    autopep8 "${check_targets[@]}" -r --diff --exit-code
+
+    # Detect invalid escape sequences in docstrings.
+    # To workaround Travis issue (https://github.com/travis-ci/travis-ci/issues/7261),
+    # ignore DeprecationWarning raised in `site.py`.
+    python -Werror::DeprecationWarning -Wignore::DeprecationWarning:site -m compileall -f -q "${check_targets[@]}"
+}
+
+
+step_chainer_install_from_sdist() {
+    # Build sdist.
+    # sdist does not support out-of-source build.
+    pushd "$REPO_DIR"
+    python setup.py sdist
+    popd
+
+    # Install from sdist
+    local envs=(MAKEFLAGS=-j"$DEFAULT_JOBS")
+
+    if [[ $SKIP_CHAINERX != 1 ]]; then
+        envs+=(CHAINER_BUILD_CHAINERX=1)
+    fi
+    env "${envs[@]}" pip install "$REPO_DIR"/dist/*.tar.gz
+}
+
+
+step_chainer_tests() {
+    pytest -m "not slow and not gpu and not cudnn and not ideep" "$REPO_DIR"/tests/chainer_tests
+}
+
+
+step_chainermn_tests() {
+    for NP in 1 2; do
+        OMP_NUM_THREADS=1 \
+            mpiexec -n ${NP} pytest -s -v -m 'not gpu and not slow' "$REPO_DIR"/tests/chainermn_tests
+    done
+}
+
+
+step_docs() {
+    make -C "$REPO_DIR"/docs html;
+}
+
+
+step_before_install_chainerx_style_check_deps() {
+    [ $TRAVIS_OS_NAME = "linux" ]  # currently only tested in linux
+
+    sudo apt-get install -y \
+         clang-format-6.0 \
+         parallel \
+
```

Comment:
```suggestion
```

Reply:
That would be a syntax error.

Reply:
A minor but how about writing it on a single line?

Reply:
Breaking into multiple lines like this is preferred.
* Easier to write comments on each item.
* Easier to understand it's a list.
* Diff will be minimized if partly modified.

Reply:
Okay, let's keep the current code.